src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -591,12 +617,23 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                             obligation.parent_trait_ref.skip_binder().print_only_trait_path(),
                         ),
                     );
-                    err.span_suggestion(
-                        span,
-                        "consider borrowing here",
-                        format!("&{}", snippet),
-                        Applicability::MaybeIncorrect,
-                    );
+
+                    // This if is to prevent a special edge-case
+                    if !has_colon || has_double_colon || has_bracket {
```

Comment:
I would prefer this to check for `!span.from_expansion()` than using snippets.

Reply:
uhhh thanks. That's much better :)